### ISSUE TITLE
Assets replaced with same filename don't trigger cache clearing

### ISSUE THREAD

**Issue by roberttolton** [aelvan/Imager-Craft#52]
As per our discussion on Slack, an assets (image, in my tests) uploaded and replaced over an existing assets with the same filename does not trigger any Imager cache clearing functions, thus the Imager transformed cached image on the front-end does not get updated.

**Comment by bossanova808** [comment]
Pretty sure we have seen this too.

**Comment by aelvan** [comment]
Forgot to write an update on this. 

There is support for the `assets.onReplaceFile` event in Imager and has been for a while. I assumed that this also accounted for when you drag a file into Assets, and choose "Replace" when the folder already has a file with that name, not only when you choose "Replace file" from the element action menu. But, according to P&T, these are two different things: the latter is actually replacing the file on disk, whereas the first actually replaces the asset (with fields and everything). 

There isn't a specific event for the first, because that's basically deleting one asset, and creating a new one. I think the only solution would be to actually delete transforms when an asset is deleted. I opted not to do this when I created Imager because that would mean that cached pages that used (non-cached) transforms that got deleted, would show broken images until the cache was updated. Not sure which scenario is the worst atm.
